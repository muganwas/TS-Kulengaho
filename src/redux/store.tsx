import { combineReducers, applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { apiMiddleware } from 'redux-api-middleware';
import logger from 'redux-logger';

var middleware;
// apply logger only in development and staging
if (process.env.NODE_ENV !== 'production') middleware = applyMiddleware(promise, thunk, apiMiddleware, logger);
else middleware = applyMiddleware(promise, thunk, apiMiddleware);

const initialState = {};

const allReducers = combineReducers({
//put all reducers here
});

const store = createStore(
    allReducers,
    initialState,
    middleware 
)

export default store;