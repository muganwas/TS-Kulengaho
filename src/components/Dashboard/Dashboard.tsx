import React, { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { setUserLogInStatus } from '../../redux/actions'
import { useHistory } from 'react-router-dom'
import { RootState } from '../../sharedTypes'
import Header from '../Header'

const Dashboard: React.FC = () => {
	const history = useHistory()
	const dispatch = useDispatch()
	const userInfo = useSelector((store: RootState) => store.userInfo)

	useEffect(() => {
		window.addEventListener('popstate', () => {
			history.push('/Dashboard')
		})
	}, [history])

	useEffect(() => {
		if (history) {
			const state: any = history.location.state
			if (state && state.token) {
				dispatch(setUserLogInStatus(true))
				return
			}
			if (!userInfo.is_logged_in && state && !state.token) {
				history.push('/')
				return
			}
		}
	}, [history, userInfo.is_logged_in, dispatch])

	return (
		<div id='dashboard-container' data-testid='dashboard-container'>
			<Header title={'Kulengaho Dash'} />
			<h1>Dash</h1>
		</div>
	)
}
export default Dashboard
