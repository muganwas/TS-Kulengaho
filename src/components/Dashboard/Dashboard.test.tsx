import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from '../../redux/store'
import Dashboard from './Dashboard'

describe('Dashboard tests', () => {
	it('renders dash container correctly', () => {
		render(
			<Provider store={store}>
				<Dashboard />
			</Provider>
		)
		const container = screen.getAllByTestId('dashboard-container')
		expect(container).toBeDefined()
		expect(container).toMatchSnapshot()
	})
})
