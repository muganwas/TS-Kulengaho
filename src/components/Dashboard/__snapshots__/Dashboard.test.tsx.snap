// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard tests renders dash container correctly 1`] = `
Array [
  <div
    data-testid="dashboard-container"
    id="dashboard-container"
  >
    <div
      class="header-container"
      data-testid="header-container"
    >
      <div
        class="logo-title-container"
      >
        <img
          alt="logo"
          class="logo"
          data-testid="logo"
          src="logo.svg"
        />
        <div
          class="header-text"
        >
          Kulengaho Dash
        </div>
      </div>
      <div
        class="sub-title"
      >
        Kulengaho
      </div>
    </div>
    <h1>
      Dash
    </h1>
  </div>,
]
`;
